perfdash/parser.go

```diff
@@ -49,24 +49,28 @@ func parseTestOutput(scanner *bufio.Scanner, job string, tests Tests, buildNumbe
 				continue
 			}
 			state = inTest
-			if _, found := result[testName]; !found {
-				result[testName] = BuildData{Job: job, Builds: map[string][]perftype.DataItem{}}
-			} else if result[testName].Job != job {
+			if _, found := result[testName]; found && result[testName].Job != job {
 				// If the job is different, we'll skip the test and keep the old result
 				state = scanning
 			}
 		}
 		if state == processing {
-			// TODO(random-liu): This is brittle, we should emit a tail delimiter too
-			if strings.Contains(line, "INFO") || strings.Contains(line, "STEP") || strings.Contains(line, "Failure") || strings.Contains(line, "[AfterEach]") {
+			// The perf data is end with an empty line.
```

Comment:
@gmarek There won't be empty line in json format. And even if there is some empty line accidentally, the loop will keep trying. :)